src/Akka.Streams.Kafka/Stages/Consumers/Abstract/BaseSingleSourceLogic.cs

```diff
@@ -118,45 +118,35 @@ protected virtual void MessageHandling((IActorRef, object) args)
                         _buffer.Enqueue(consumerMessage);
                     
                     Pump();
-                    
                     break;
                 
                 case Status.Failure failure:
-                    var exception = failure.Cause;
-                    switch (_decider(failure.Cause))
-                    {
-                        case Directive.Stop:
-                            // Throw
-                            FailStage(exception);
-                            break;
-                        case Directive.Resume:
-                            // keep going
-                            break;
-                        case Directive.Restart:
-                            // TODO: Need to do something here: https://github.com/akkadotnet/Akka.Streams.Kafka/issues/33
-                            break;
-                    }
+                    FailStage(failure.Cause);
                     break;
                 
-                case Terminated terminated:
+                case Terminated terminated when ReferenceEquals(terminated.ActorRef, ConsumerActor):
```

Comment:
Why is the `ReferenceEquals` needed here?

Reply:
What other actors are being `Context.Watch`'d by this actor?

Reply:
It has to do with clean shutdown ordering

Reply:
What other actors could send a `Terminate` message though?

Reply:
This method is not being used in this class, it is being passed as the receive method of the StageActor

Reply:
If you look in the PreStart(), we explicitly told the StageActor instance to watch the KafkaConsumerActor instance.